packages/core-mobile/app/store/portfolio/slice.ts

```diff
@@ -13,17 +13,26 @@ export const portfolioSlice = createSlice({
       action: PayloadAction<{ tokenId: string }>
     ) => {
       const { tokenId } = action.payload
-      state.tokenVisibility[tokenId.toLowerCase()] =
-        !state.tokenVisibility[tokenId.toLowerCase()]
+
+      if (state.tokenVisibility?.[tokenId.toLowerCase()] != null) {
```

Comment:
we actually usually try not to use != null since it is kinda vague. it looks like you can just check against undefined